pkg/runctl/controller.go

```diff
@@ -868,3 +872,56 @@ func (c *Controller) addToWorkqueueFromAssociated(obj interface{}) {
 		)
 	}
 }
+
+func (c *Controller) finalizePipelineRun(ctx context.Context, pipelineRun k8s.PipelineRun) error {
+
+	if !pipelineRun.HasFinalizer() {
+		return nil
+	}
+
+	err := pipelineRun.DeleteFinalizerAndCommitIfExists(ctx)
+	if err != nil {
+		return err
+	}
+
+	c.logFinalState(ctx, pipelineRun)
+
+	return nil
+}
+
+func (c *Controller) logFinalState(ctx context.Context, pipelineRun k8s.PipelineRun) error {
+	status := pipelineRun.GetStatus()
+	spec := pipelineRun.GetSpec()
+	runID := "unknown"
+	if spec != nil && spec.Logging != nil && spec.Logging.Elasticsearch != nil {
+		err := error(nil)
+		runID, err = toJSONString(&spec.Logging.Elasticsearch.RunID)
+		if err != nil {
+			return errors.WithMessage(err,
+				"could not serialize spec.logging.elasticsearch.runid to JSON",
+			)
+		}
```

Comment:
Instead of aborting, it would be better to continue with `runID = "invalid"`.

Reply:
Done.